homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -1,291 +1,346 @@
 from homeassistant.components.media_player.const import (
-    MEDIA_TYPE_MUSIC,
+    REPEAT_MODE_OFF,
+    SUPPORT_CLEAR_PLAYLIST,
     SUPPORT_NEXT_TRACK,
     SUPPORT_PAUSE,
     SUPPORT_PLAY,
     SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_REPEAT_SET,
+    SUPPORT_SELECT_SOUND_MODE,
     SUPPORT_SELECT_SOURCE,
+    SUPPORT_SHUFFLE_SET,
     SUPPORT_STOP,
     SUPPORT_TURN_OFF,
     SUPPORT_TURN_ON,
     SUPPORT_VOLUME_MUTE,
     SUPPORT_VOLUME_SET,
 )
-from homeassistant.const import (
-    CONF_HOST,
-    CONF_PORT,
-    STATE_IDLE,
-    STATE_ON,
-    STATE_PAUSED,
-    STATE_PLAYING,
-    STATE_UNKNOWN,
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import MusicCastDataUpdateCoordinator, MusicCastDeviceEntity
+from .const import (
+    DEFAULT_ZONE,
+    DOMAIN,
+    HA_REPEAT_MODE_TO_MC_MAPPING,
+    MC_REPEAT_MODE_TO_HA_MAPPING,
 )
-import homeassistant.helpers.config_validation as cv
-import homeassistant.util.dt as dt_util
 
-_LOGGER = logging.getLogger(__name__)
+PARALLEL_UPDATES = 1
 
-SUPPORTED_FEATURES = (
-    SUPPORT_PLAY
-    | SUPPORT_PAUSE
-    | SUPPORT_STOP
-    | SUPPORT_PREVIOUS_TRACK
-    | SUPPORT_NEXT_TRACK
-    | SUPPORT_TURN_ON
-    | SUPPORT_TURN_OFF
+
+MUSIC_PLAYER_SUPPORT = (
+    SUPPORT_PAUSE
     | SUPPORT_VOLUME_SET
     | SUPPORT_VOLUME_MUTE
+    | SUPPORT_TURN_ON
+    | SUPPORT_TURN_OFF
+    | SUPPORT_CLEAR_PLAYLIST
+    | SUPPORT_PLAY
+    | SUPPORT_SHUFFLE_SET
+    | SUPPORT_REPEAT_SET
+    | SUPPORT_PREVIOUS_TRACK
+    | SUPPORT_NEXT_TRACK
+    | SUPPORT_SELECT_SOUND_MODE
     | SUPPORT_SELECT_SOURCE
+    | SUPPORT_STOP
 )
 
-KNOWN_HOSTS_KEY = "data_yamaha_musiccast"
-INTERVAL_SECONDS = "interval_seconds"
 
-DEFAULT_PORT = 5005
-DEFAULT_INTERVAL = 480
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up MusicCast sensor based on a config entry."""
+    coordinator: MusicCastDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-        vol.Optional(INTERVAL_SECONDS, default=DEFAULT_INTERVAL): cv.positive_int,
-    }
-)
+    name = coordinator.data.network_name
+
+    media_players: list[Entity] = []
+
+    for zone in coordinator.data.zones:
+        zone_name = name if zone == DEFAULT_ZONE else f"{name} {zone}"
+
+        media_players.append(
+            MusicCastMediaPlayer(zone, zone_name, entry.entry_id, coordinator)
+        )
 
+    async_add_entities(media_players)
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
```

Comment:
We should implement `async_setup_platform` and import the old yaml config host to a config flow. Also log a deprecation warning here, that the old config section is deprecated.

Reply:
Good idea, I implemented this so that we should have backward compatibility